File: src/main/java/com/example/demo20250620/entity/WorkRecord.java
package com.example.demo20250620.entity;

import jakarta.persistence.*;

@Entity
public class WorkRecord {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "work_time")
    private String workTime;

    @Column(name = "work_location")
    private String workLocation;

    @Column(name = "work_content")
    private String workContent;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private SysUser user;

    @Column(name = "detail")
    private String detail;

    public WorkRecord() {
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getWorkTime() {
        return workTime;
    }

    public void setWorkTime(String workTime) {
        this.workTime = workTime;
    }

    public String getWorkLocation() {
        return workLocation;
    }

    public void setWorkLocation(String workLocation) {
        this.workLocation = workLocation;
    }

    public String getWorkContent() {
        return workContent;
    }

    public void setWorkContent(String workContent) {
        this.workContent = workContent;
    }

    public SysUser getUser() {
        return user;
    }

    public void setUser(SysUser user) {
        this.user = user;
    }

    public String getDetail() {
        return detail;
    }

    public void setDetail(String detail) {
        this.detail = detail;
    }
}

File: src/main/webapp/app/model/workrecord.js
Ext.define('AM.model.workrecord',{
    extend:'Ext.data.Model',
    fields:[
        {name:'id',type:'int'},
        {name:'workTime',type:'string'},
        {name:'workLocation',type:'string'},
        {name:'workContent',type:'string'},
        {name:'user',type:'auto'},
        {name:'detail',type:'string'},
        {name:'username',type:'string', convert:function(v, record){
            var user = record.get('user');
            return user ? user.sysusername : '';
        }}
    ],
})
